PythonPractice/TicTacToe/tictactoe.py

```diff
@@ -0,0 +1,155 @@
+class SpaceTakenError(Exception):
+    pass
+
+class BoardTooSmallError(Exception):
+    pass
+
+class TicTacToe:
+
+    def __init__(self):
+        self.board = []
+        self.initialize_board()
+        self.playerO = 'O'
+        self.playerX = 'X'
+        self.player = self.playerX
+
+    def initialize_board(self):
+        """
+        Creates an empty board to be played on
+        """
+        # get board size from user
+        good_input = False
+        size = None
+        while not good_input:
+            try:
+                size = int(raw_input("What size board to do you want to play on?\n"))
+                if size < 3:
+                    raise BoardTooSmallError
+            except BoardTooSmallError:
+                print "Your board must be at least a 3x3 to play."
+            except ValueError:
+                print "Your board size must be an integer."
+            else:
+                good_input = True
+        # initialize board to empty strings
+        self.board = [[None for row in range(0, size)] for col in range(0, size)]
```

Comment:
You can drop the zero.  `range(5)` == `range(0, 5)` == `[0, 1, 2, 3, 4]`.

Reply:
Also, this is another nitpicky Python 2.x vs. 3.x thing, but in 2.x, most of the time you want to use `xrange` instead of `range`.  In 2.x, the latter actually generates a list in memory while the former just pretends to.  It'd take a pretty extreme case for this to make a practical impact, but the difference is still worth knowing.